### PR TITLE
Priority Queue sea_is_modified PR 1

### PR DESCRIPTION
**Pull request by achenchen1**

### CODE REVIEW

File: seahorn/jobs2/priority_queue_capacity2/sea.yaml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+verify_options:
+  horn-bv2-tracking-mem: true
```
Comment (achenchen1) [inline seahorn/jobs2/priority_queue_capacity2/sea.yaml:2]: Github isn't showing this, but I have a newline here already. Ditto for the other sea.yaml files.

File: seahorn/jobs2/priority_queue_capacity2/aws_priority_queue_capacity_harness2.c [hunk 2]
```diff
@@ -12,38 +12,29 @@ int main(void) {
...
-  save_byte_from_array((uint8_t *)old_container.data,
-                       old_container.current_size, &old_byte_container);
-
-  /* save current state of the backpointers structure */
-  struct aws_array_list old_backpointers = queue.backpointers;
-  struct store_byte_from_buffer old_byte_backpointers;
-  save_byte_from_array((uint8_t *)old_backpointers.data,
-                       old_backpointers.current_size, &old_byte_backpointers);
+  sea_tracking_on();
+  sea_reset_modified((char *)&queue.container);
```
Comment (achenchen1) [inline seahorn/jobs2/priority_queue_capacity2/aws_priority_queue_capacity_harness2.c:22]: If my understanding is correct, also based on example from byte_buf, this should be correct?

File: seahorn/jobs2/priority_queue_capacity2/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,8 @@
+add_executable(
+  priority_queue_capacity2
+  ${AWS_C_COMMON_ROOT}/source/array_list.c
+  ${AWS_C_COMMON_ROOT}/source/priority_queue.c
+  aws_priority_queue_capacity_harness2.c
+)
+sea_attach_bc_link(priority_queue_capacity2)
+sea_add_unsat_test(priority_queue_capacity2)
```
Comment (achenchen1) [inline seahorn/jobs2/priority_queue_capacity2/CMakeLists.txt:9]: Removed fuzz, smack, and KLEE tests from all the copied CMakeLists.txt

File: seahorn/jobs2/priority_queue_capacity2/aws_priority_queue_capacity_harness2.c (hunk 2 above)
Comment (priyasiddharth) [reply seahorn/jobs2/priority_queue_capacity2/aws_priority_queue_capacity_harness2.c:22]: I am not sure. Why prefer this over comparing `queue.container` field by field.

File: seahorn/jobs2/priority_queue_size2/aws_priority_queue_size_harness2.c [hunk 4]
```diff
@@ -14,31 +14,24 @@ int main(void) {
...
-    struct store_byte_from_buffer old_byte_container;
-    save_byte_from_array((uint8_t *)old_container.data, old_container.current_size, &old_byte_container);
-
-    /* save current state of the backpointers structure */
-    struct aws_array_list old_backpointers = queue.backpointers;
-    struct store_byte_from_buffer old_byte_backpointers;
-    save_byte_from_array((uint8_t *)old_backpointers.data, old_backpointers.current_size, &old_byte_backpointers);
+    /* save state */
+    sea_tracking_on();
+    sea_reset_modified((char *)&queue.container);
```
Comment (priyasiddharth) [inline seahorn/jobs2/priority_queue_size2/aws_priority_queue_size_harness2.c:22]: same comment as above for queue container

File: seahorn/jobs2/priority_queue_top2/aws_priority_queue_top_harness2.c [hunk 5]
```diff
@@ -15,37 +15,27 @@ int main(void) {
...
-  save_byte_from_array((uint8_t *)old_container.data,
-                       old_container.current_size, &old_byte_container);
-
-  /* save current state of the backpointers structure */
-  struct aws_array_list old_backpointers = queue.backpointers;
-  struct store_byte_from_buffer old_byte_backpointers;
-  save_byte_from_array((uint8_t *)old_backpointers.data,
-                       old_backpointers.current_size, &old_byte_backpointers);
+  sea_tracking_on();
+  sea_reset_modified((char *)&queue.container);
```
Comment (priyasiddharth) [inline seahorn/jobs2/priority_queue_top2/aws_priority_queue_top_harness2.c:24]: same comment as above

**Review by priyasiddharth — CHANGES_REQUESTED** [review]
For all harnesses, can you explain why `queue.container` and `queue.backpointers` cannot be compared field-by-field?

1. If no reason emerges, I would advise comparing field by field over using `is-modified`. The reason is that `is-modified` takes a pointer which forces the compiler to generate the struct in RAM. If no one (like `is-modified`) relies on address being used then the struct need not have an address and the compiler can instead promote the struct to be distributed across registers.
2. In case of buffers, the buffers will necessarily be generated in memory since the only way to access a buffer is my read/write to a memory location. Thus `is-modified` does not move an otherwise register access to memory.

**Pull request closed without merging**